### PR TITLE
adding polygon to raster map

### PR DESCRIPTION
**Pull request by lindsaycarr**
_Please don't merge yet!_

Still not perfect....without adding the `sp` polygon, the raster map looks like this:

image

but then when I add it, it looks like this:
image

There seems to be an issue with the projections because of the lat/long values are so different.

```
> sp::bbox(precip_sp_projected)
  min max
x   1  21
y   1  29
> sp::bbox(projected_sp)
      min       max
x  515163  554718.2
y 2252248 2304876.6
```

### CODE REVIEW

**Comment by wdwatkins** [comment]
@lindsaycarr `projected_sp` has to be in UTM coordinates, those aren't lat/lons

**Comment by lindsaycarr** [comment]
@wdwatkins yeah, I'm not really sure what that means or how to do that?

**Comment by wdwatkins** [comment]
Hmm, also the min/max you are getting for the raster I think is just the raster cell coordinates (ie it is 21x29 pixels).  If you look at a map that won't be anywhere near WI.  

Maybe see look into how the base raster is georeferenced?

**Comment by wdwatkins** [comment]
It looks fairly simple to transform from UTM to lat/lon if you need to.  https://stackoverflow.com/questions/36520915/converting-utms-to-lat-long-in-r

**Comment by lindsaycarr** [comment]
OK, so I looked at this page (https://gis.stackexchange.com/questions/23841/create-a-raster-with-georeferenced-information-in-r) and realized I had the exact extent of the raster in order to georeference it to lat/long, so I added that step. I'm much closer because the watershed polygon plots over the same region, but now the precip values don't show up:

image

**Comment by wdwatkins** [comment]
Good progress, is there a white background with the polygon maybe?

**Comment by lindsaycarr** [comment]
There was something strange happening with re-projecting it. Something about warping. I kept it in the projection string that geoknife automatically uses and it works:

image

**Comment by lindsaycarr** [comment]
@wdwatkins can you do another quick review? I can run make and this is what I get for the clip fig: 

image

**Comment by wdwatkins** [comment]
If you're dealing with cosmetic stuff yet I would say reduce the color scale range so it is all used, unless there will be other maps with values that go to two.  Will look at the code later this morning.

**Comment by lindsaycarr** [comment]
Well, the maximum value is 1.8 ... so I had the scale round up from the maximum. What's the best way to do it?

**Comment by wdwatkins** [comment]
Oh ok, never mind then.  It must just be a single cell or something.

**Comment by lindsaycarr** [comment]
They all look faded though, like alpha is not 1. I haven't figured that out.

File: scripts/visualize/visualize.raster_map_precip.R [hunk 1]
```diff
@@ -1,15 +1,48 @@
...
+    #crop the precip data to the watershed polygon
+    precip_sp <- raster::crop(precip_sp, geom_feature)
+  }
+  
+  precip_sp_df <- as.data.frame(precip_sp) # prep sp data
+  names(precip_sp_df) <- c("value", "x", "y")
+  
+  map_plot <- ggplot() + 
+    ggplot2::geom_tile(data = precip_sp_df,
+                       ggplot2::aes(x=x, y=y, fill = value, alpha=ifelse(is.na(value), 0, 1))) +
```
Comment (wdwatkins) [inline scripts/visualize/visualize.raster_map_precip.R]: If there is something up with the alpha values, could this be it?  Can you still put in function calls with the non-standard eval here?

**Comment by lindsaycarr** [comment]
Hmm yep - that's the issue. I thought I had that working for something previously? I'll submit changes to that and then merge this! Thanks!

**Comment by lindsaycarr** [comment]
Here's what it looks like now (still not included in publish)

image

**Pull request merged**